test/core/Safe.GuardManager.spec.ts

```diff
@@ -17,63 +17,66 @@ import { chainId } from "../utils/encoding";
 
 describe("GuardManager", async () => {
     const [user1, user2] = waffle.provider.getWallets();
+    const GUARD_STORAGE_SLOT = ethers.utils.keccak256(ethers.utils.toUtf8Bytes("guard_manager.guard.address"));
 
     const setupWithTemplate = deployments.createFixture(async ({ deployments }) => {
         await deployments.fixture();
-        const mock = await getMock();
+        const validGuardMock = await getMock();
 
         const guardContract = await hre.ethers.getContractAt("Guard", AddressZero);
-        const guardEip165Calldata = guardContract.interface.encodeFunctionData("supportsInterface", ["0xe6d7a83a"]);
-        await mock.givenCalldataReturnBool(guardEip165Calldata, true);
+        const guardEip165Calldata = guardContract.interface.encodeFunctionData("supportsInterface", ["0x945b8148"]);
+        await validGuardMock.givenCalldataReturnBool(guardEip165Calldata, true);
         const safe = await getSafeWithOwners([user2.address]);
-        await executeContractCallWithSigners(safe, safe, "setGuard", [mock.address], [user2]);
+        await executeContractCallWithSigners(safe, safe, "setGuard", [validGuardMock.address], [user2]);
+
         return {
             safe,
-            mock,
+            validGuardMock,
             guardEip165Calldata,
         };
     });
 
     describe("setGuard", async () => {
```

Comment:
The tests were a little bit overcomplicated. A test that tests one thing would implicitly test another in the test body. I had to break them up a little bit